src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -255,10 +257,19 @@ export function getInputType(
 
     function getInputModelType(m: Model): InputType {
         const intrinsicName = getIntrinsicModelName(program, m);
-        if (intrinsicName && intrinsicName === "string") {
-            const values = getKnownValues(program, m);
-            if (values) {
-                return getInputModelForExtensibleEnum(m, values);
+        if (intrinsicName) {
+            switch (intrinsicName) {
```

Comment:
Shall we  wrap all the model type who has intrinsicName as intrinsic type? need to discuss with arch board.

Reply:
I think we should unbox all types, just like a compiler does.